src/lib/core/ExchangeContext.cpp

```diff
@@ -524,10 +543,16 @@ WEAVE_ERROR ExchangeContext::SendMessage(uint32_t profileId, uint8_t msgType, Pa
         msgInfo->Flags |= kWeaveMessageFlag_ReuseMessageId;
     if (sendFlags & kSendFlag_ReuseSourceId)
         msgInfo->Flags |= kWeaveMessageFlag_ReuseSourceId;
-    if (sendFlags & kSendFlag_MulticastFromLinkLocal)
-        msgInfo->Flags |= kWeaveMessageFlag_MulticastFromLinkLocal;
+    if (sendFlags & kSendFlag_DefaultMulticastSourceAddress)
+        msgInfo->Flags |= kWeaveMessageFlag_DefaultMulticastSourceAddress;
+
+    SetFlag(msgInfo->Flags, kWeaveMessageFlag_FromInitiator, IsInitiator());
+
+#if WEAVE_CONFIG_ENABLE_EPHEMERAL_UDP_PORT
```

Comment:
I would have expected this conditional wrapper in more places, for example lines 330 and possibly in UDPEndpoint.cpp. Any reason it isn't used in those locations?

Reply:
I went back and forth on this.  I agree that SetUseEphemeralUDPPort() should be conditionalized and will make that change.  However I'm inclined to leave UseEphemeralUDPPort() as an inline that always returns false.

Regarding UDPEndPoint, ephemeral ports are a feature of the IP stack and thus have always been supported in UDPEndPoint.  The only related change to UDPEndPoint is the new GetBoundPort() method which is used (and useful) even when support for ephemeral UDP ports is turned off in Weave.